Modules/LearningSequence/classes/Player/class.ilLegacyKioskModeView.php

```diff
@@ -68,12 +67,33 @@ public function buildInitialState(State $empty_state): State
 	 */
 	public function buildControls(State $state, ControlBuilder $builder): ControlBuilder
 	{
-		$builder->start (
-			$this->lng->txt('lso_start_item').' '.$this->getTitleByType($this->getType()),
-			self::CMD_START_OBJECT,
-			0
+		if(! $builder instanceof LSControlBuilder) {
+			throw new LogicException("The Legacy Mode in the Learning Sequence requires an LSControlBuilder explicitely.", 1);
+		}
+
+		$label = $this->lng->txt('lso_start_item').' '.$this->getTitleByType($this->getType());
+
+		$ref_id = $this->object->getRefId();
+		$type = $this->object->getType();
+
+		$url = \ilLink::_getStaticLink(
+			$ref_id,
+			$type,
+			true,
+			false
 		);
 
+		if(in_array($type, self::GET_VIEW_CMD_FROM_LIST_GUI_FOR)) {
+			$obj_id = $this->object->getId();
+			$item_list_gui = \ilObjectListGUIFactory::_getListGUIByType($type);
+			$item_list_gui->initItem($ref_id, $obj_id);
+			$view_link = $item_list_gui->getCommandLink('view');
+			$view_link = str_replace('&amp;', '&', $view_link);
+			$view_link = ILIAS_HTTP_PATH.'/'.$view_link;
+			$url = $view_link;
+		}
+
+		$builder->start($label,	$url, 0);
 		//return $this->debugBuildAllControls($builder);
```

Comment:
```suggestion
```

Reply:
I'd rather leave this; otherwise, we probably should delete the method as well.